### ISSUE TITLE
Remove Race Condition

### ISSUE THREAD

**Issue by ghunkins** [jqphu/PocketUniverse#19]
The below lines introduce a race condition based on the storage of the simulation. The resolution of `fetchSimulate` relies on the simulation already being added to `chrome.storage.sync`, so they cannot be run in parallel.

https://github.com/jqphu/PocketUniverse/blob/ff3cbc0b346e7c6d0217df1efefbcc7ca13d30aa/src/lib/storage.ts#L140-L146

**Comment by ghunkins** [comment]
I might have opened this prematurely. Looks like `fetchSimulate` does not in fact use `chrome.storage.sync`. Taking a closer look at the code now.

**Comment by jqphu** [comment]
Thanks for taking a look!

At first glance I thought this was a bug too! But I think `fetchSimulate` just gets the data in `response` then we update the storage afterwards.

**Comment by ghunkins** [comment]
Absolutely. All looks good on closer look, confused it with another issue (my end), apologies for the false alarm!!

**Comment by jqphu** [comment]
No worries - appreciate you looking into it either way!
